# HG changeset patch
# User Rik <rik@octave.org>
# Date 1532459171 25200
#      Tue Jul 24 12:06:11 2018 -0700
# Node ID e189ee5cd537af6c1a80627edcd4a55121495d54
# Parent  4de7535ece1119cc71d6c1e843e010bf22e8ea91
io.tst: Clear global variable a1 at end of test to avoid leaking vars.

* io.tst (testls): Clear global variable a1 created in subfunction at end of
function.

diff --git a/test/io.tst b/test/io.tst
--- a/test/io.tst
+++ b/test/io.tst
@@ -172,16 +172,19 @@
 %!          || isglobal ("a13") || isglobal ("a14") || isglobal ("a15")
 %!          || isglobal ("a16") || isglobal ("a17") || isglobal ("a18")
 %!          || isglobal ("a19") || isglobal ("a20"))
 %!        error ("failed: %s global test", file);
 %!      endif
 %!    endfor
 %!  endif
 %!
+%!  ## Cleanup after test
+%!  clear -global a1;
+%!
 %!  ret = 1;
 %!endfunction
 
 %!testif HAVE_ZLIB
 %!
 %! [save_status, save_files] = testls (0);
 %! [load_status, load_files] = testls (1);
 %!
